skfs/src/Context.sk

```diff
@@ -1626,4 +1626,31 @@ fun withRegionValue<T>(f: () ~> T): T {
   copied.getHead();
 }
 
+fun withRegionIterate<K, T>(
+  it: mutable Iterator<K>,
+  value: T,
+  fn: (K, T) ~> T,
+): T {
+  optSaved = Some(SKStore.newObstack());
+  valueInRegion = value;
+  loop {
+    saved = optSaved.fromSome();
+    it.next() match {
+    | Some(k) -> !valueInRegion = fn(k, valueInRegion)
+    | _ -> break void
+    };
+    if (shouldGC(saved) != 0) {
+      !valueInRegion = SKStore.destroyObstackWithValue(
+        saved,
+        List[value],
+      ).getHead();
+      !optSaved = Some(SKStore.newObstack());
+    }
+  };
+  SKStore.destroyObstackWithValue(
+    optSaved.fromSome(),
+    List[valueInRegion],
+  ).getHead();
```

Comment:
withRegion protects against exceptions, I guess this ought to in a similar way right?

Reply:
Yes please. This is my main reason for wanting to replace my manual obstack code in write-csv with something like this.